test-infra/camel-test-infra-kafka/src/test/java/org/apache/camel/test/infra/kafka/services/ContainerLocalSingletonKafkaService.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.camel.test.infra.kafka.services;
+
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.testcontainers.containers.KafkaContainer;
+import org.testcontainers.utility.DockerImageName;
+
+public class ContainerLocalSingletonKafkaService extends ContainerLocalKafkaService
+        implements ExtensionContext.Store.CloseableResource {
+    private static boolean started;
+
+    public ContainerLocalSingletonKafkaService(KafkaContainer container) {
+        super(container);
+    }
+
+    public ContainerLocalSingletonKafkaService() {
+        super();
+    }
+
+    @Override
+    public void beforeAll(ExtensionContext extensionContext) {
+        if (!started) {
+            started = true;
+            extensionContext.getRoot().getStore(ExtensionContext.Namespace.GLOBAL).put("kafka", this);
+            super.initialize();
+        }
```

Comment:
It is not really thread safe, in case of parallel build we could face issues with it. To simplify and make it thread safe, you could use `getOrComputeIfAbsent` instead to initialize it only once, like I did [here](https://github.com/apache/camel/blob/main/components/camel-test/camel-test-main-junit5/src/main/java/org/apache/camel/test/main/junit5/CamelMainExtension.java#L76)

Reply:
Hmm I doubt these ITs were run in parallel. That would mean that multiple Kafka instances were started in parallel :)

But in the spirit of running these in parallel in the future I'll add the suggested changes :) thanks !

Reply:
I maybe wrong but AFAIK, the test-infra artifacts are part of the Camel API for the tests which means that it is meant to be used by the camel community so it could potentially be used in a parallel build even if we don't do it within the context of Camel itself. 

Here as you tied it to the `GLOBAL` namespace, if it is thread safe (by using `getOrComputeIfAbsent` for example), it should start only one Kafka instance even in the context of a parallel build